src/state/hooks.ts

```diff
@@ -82,13 +82,66 @@ export const useFarmUser = (pid) => {
   }
 }
 
-export const useLpTokenPrice = (symbol: string) => {
-  const farm = useFarmFromSymbol(symbol)
-  const tokenPriceInUsd = useGetApiPrice(getAddress(farm.token.address))
+// Return a farm for a given token symbol. The farm is filtered based on attempting to return a farm with a quote token from an array of preferred quote tokens
+export const useFarmFromTokenSymbol = (tokenSymbol: string, preferredQuoteTokens?: string[]): Farm => {
+  const farms = useSelector((state: State) => state.farms.data.filter((farm) => farm.token.symbol === tokenSymbol))
```

Comment:
Can you use `useFarmFromSymbol`?

Reply:
`useFarmFromSymbol` was just returning the first farm that it found in the constants, so the order in which the config is maintained becomes important. i.e. [with this example](https://github.com/pancakeswap/pancake-frontend/pull/1244#discussion_r634528092) if the `KUN-QSD` farm came before `KUN-BUSD` in farms.ts, it would try to retrieve the price based on the QSD pair, and it wouldn't be able to get a price.

The new filtering logic means it would use the `KUN-BUSD` farm, even if `KUN-QSD` came before it in the config.